test/test_nn.py

```diff
@@ -13783,6 +13783,37 @@ def test_MaxPool_zero_batch_dim(self, device):
             inp = torch.ones(1, 0, 50, 44, 31, device=device)
             mod(inp)
 
+    @onlyOnCPUAndCUDA
+    def test_MaxUnpool_zero_batch_dim(self, device):
+        pool = torch.nn.MaxPool1d(2, stride=2, return_indices=True).to(device)
+        unpool = torch.nn.MaxUnpool1d(2, stride=2).to(device)
+        inp = torch.randn(0, 10, 10, requires_grad=True, device=device)
+        output, indices = pool(inp)
+        output.requires_grad_(True)
+        unpool_out = unpool(output, indices)
+        unpool_out.sum().backward()
+
+        self.assertEqual(inp.grad, torch.zeros_like(inp))
```

Comment:
I think both grads and outputs should be checked here and below.

Also, probably outside of the scope of this PR, but it seems gradgrad checks aren't being done and that led to https://github.com/pytorch/pytorch/issues/64306#issuecomment-910369537. At some point, we should add testing for this

Reply:
OK thanks. Will add the tests and update.

Reply:
On a side note, I'm checking for the same things as the `_test_module_empty_input` function is checking (it does not check the outputs). Do you want to add output checks to the that function too?

Reply:
Yes please :)

Reply:
Couldn't `_test_module_empty_input` be invoked here instead of replicating the checks? If not, maybe it could be refactored slightly. Ideally it'd be nice if it was flexible enough to be used consistently throughout the zero batch size tests

Reply:
Yeah true that. Should I make an issue and add it as a refactoring step later?